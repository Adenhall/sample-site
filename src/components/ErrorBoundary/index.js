/* @flow */

import * as React from 'react';

type Props = { children?: React.Node };

type State = {
  error: Error | null,
  errorInfo: { componentStack: string } | null
};

export default class ErrorBoundary extends React.Component<Props, State> {
  // eslint-disable-next-line react/static-property-placement
  static defaultProps = {
    children: null
  };

  constructor(props: Props) {
    super(props);
    this.state = { error: null, errorInfo: null };
  }

  componentDidCatch(error: Error, errorInfo: { componentStack: string }) {
    // Catch errors in any components below and re-render with error message
    this.setState({
      error,
      errorInfo
    });
    // You can also log error messages to an error reporting service here
  }

  render(): React.Node {
    const { errorInfo, error } = this.state;
    const { children } = this.props;
    if (errorInfo) {
      // Error path
      return (
        <div>
          <h2>Something went wrong.</h2>
          <details style={{ whiteSpace: 'pre-wrap' }}>
            {error && error.toString()}
            <br />
            {errorInfo.componentStack}
          </details>
        </div>
      );
    }
    // Normally, just render children
    return children || null;
  }
}
