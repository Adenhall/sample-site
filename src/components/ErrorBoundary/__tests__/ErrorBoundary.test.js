import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import ErrorBoundary from '../index';

describe('<ErrorBoundary />', () => {
  it('renders', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <ErrorBoundary />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
