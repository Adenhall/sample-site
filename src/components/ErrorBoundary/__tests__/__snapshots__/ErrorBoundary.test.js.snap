// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> renders 1`] = `null`;
